aptos-move/block-executor/src/executor.rs

```diff
@@ -235,14 +246,16 @@ where
             *worker_idx = (*worker_idx + 1) % post_commit_txs.len();
 
             // Committed the last transaction, BlockSTM finishes execution.
-            if txn_idx as usize + 1 == scheduler.num_txns() as usize {
+            let next_tid = provider.txn_index_right_after(txn_idx);
```

Comment:
This would be cleaner if the APIs just returns `None` if there is no next txn index.